sparse_strips/vello_cpu/src/render.rs

```
@@ -448,9 +449,19 @@ impl RenderContext {
     }
 
     /// Render the current context into a pixmap.
+    /// 
+    /// Note that the pixmap needs to have the same dimensions as the render context.
     pub fn render_to_pixmap(&self, pixmap: &mut Pixmap) {
         let width = pixmap.width();
         let height = pixmap.height();
+        
+        if width != self.width || height != self.height {
```

Elsewhere, when invariants like this aren't obeyed, we typically panic:

https://github.com/linebender/vello/blob/main/sparse_strips%2Fvello_common%2Fsrc%2Fpixmap.rs#L43-L47

I'm wondering whether we should either return an Error or similarly panic (to at least be consistent).

---

Good question, I’m not sure.

---

I think the best option is allowing the consumer to decide what to do when _they_ break an invariant (i.e. return an Error). Vello hybrid can return an Error type IIRC when spatiotemporal allocation suffers slot exhaustion.

Otherwise, I think it's fine to defer that to a larger review of this question and simply panic here (to be consistent with the rest of the system).